src/main/java/org/jabref/logic/util/io/FileNameUniqueness.java

```diff
@@ -41,4 +47,65 @@ public static String getNonOverWritingFileName(Path targetDirectory, String file
 
         return newFileName;
     }
+
+    /**
+     *
+     * @param directory The directory which saves the files (.pdf, for example)
+     * @param fileName Suggest name for the newly downloaded file
+     * @param dialogService To display the error and success message
+     * @return true when the content of files is duplicate and successfully delete the newly downloaded file ,
+     *         false when it is not a duplicate file or fail to delete the duplicate file
+     * @throws IOException Fail when the file is not exist or something wrong when reading the file
+     */
+    public static boolean isDuplicatedFile(Path directory, String fileName, DialogService dialogService) throws IOException {
+
```

Comment:
Add some non-null checks - the method won't work if any parameter is null:

```suggestion

        Objects.requireNonNull(directory);
        Objects.requireNonNull(filename);
        Objects.requireNonNull(diaglogService);
```